src/main/java/com/conveyal/r5/profile/PerTargetPropagater.java

```diff
@@ -0,0 +1,107 @@
+
+    /** Times at transit stops for each iteration */
+    public final int[][] travelTimesToStopsEachIteration;
+
+    /** Times at targets using the street network */
+    public final int[] nonTransferTravelTimesToTargets;
+
+    /** The maximum allowable travel time in this regional analysis, to avoid propagating things that won't be useful */
+    public final int cutoffSeconds;
+
+    /** The linked targets */
+    public final LinkedPointSet targets;
+
+    /** the profilerequest (used for walk speed etc.) */
+    public final ProfileRequest request;
+
+    public PerTargetPropagater (int[][] travelTimesToStopsEachIteration, int[] nonTransferTravelTimesToTargets, LinkedPointSet targets, ProfileRequest request, int cutoffSeconds) {
+        this.travelTimesToStopsEachIteration = travelTimesToStopsEachIteration;
+        this.nonTransferTravelTimesToTargets = nonTransferTravelTimesToTargets;
+        this.targets = targets;
+        this.request = request;
+        this.cutoffSeconds = cutoffSeconds;
+    }
+
+    public void propagate (Reducer reducer) {
+        targets.makePointToStopDistanceTablesIfNeeded();
+
+        long startTimeMillis = System.currentTimeMillis();
+        // avoid float math in loop below
+        // float math was previously observed to slow down this loop, however it's debatable whether that was due to
+        // casts, the operations themselves, or the fact that the operations were being completed with doubles rather
+        // than floats.
+        int speedMillimetersPerSecond = (int) (request.walkSpeed * 1000);
+
+        for (int targetIdx = 0; targetIdx < targets.size(); targetIdx++) {
+            boolean[] perIterationResults = new boolean[travelTimesToStopsEachIteration.length];
+            // fill with non-transit time as upper bound on travel time
+            Arrays.fill(perIterationResults, nonTransferTravelTimesToTargets[targetIdx] < cutoffSeconds);
+
+            if (perIterationResults[0]) {
+                // reachable on foot, no need to even check if it's reachable via transit
+                reducer.accept(targetIdx, perIterationResults);
+                continue;
+            }
+
+            TIntIntMap pointToStopDistanceTable = targets.pointToStopDistanceTables.get(targetIdx);
+
+            // effectively final nonsense
+            boolean[] targetEverReached = new boolean[] { nonTransferTravelTimesToTargets[targetIdx] <= cutoffSeconds };
+
+            // don't try to propagate transit if there are no nearby transit stops,
+            // but still call the reducer below with the non-transit times, because you can walk even where there is no
+            // transit
+            if (pointToStopDistanceTable != null) {
+                for (int iteration = 0; iteration < perIterationResults.length; iteration++) {
+                    final int effectivelyFinalIteration = iteration;
+                    pointToStopDistanceTable.forEachEntry((stop, distanceMm) -> {
```

Comment:
Mm would be mega-meters, which is why I break with camelCase variable naming convention and put underscores before unit names (to preserve capitalization) or spell out the unit prefixes.